lib/node_modules/@stdlib/_tools/eslint/rules/tsdoc-doctest/lib/main.js

```diff
@@ -0,0 +1,487 @@
+	}
+	return regexCache[ key ];
+}
+
+/**
+* Counts the number of lines in the given string.
+*
+* @private
+* @param {string} str - input string
+* @returns {number} number of lines
+*/
+function countLines( str ) {
+	var matches;
+
+	// Use cached result if available
+	if ( lineCountCache[ str ] ) {
+		return lineCountCache[ str ];
+	}
+	matches = str.match( RE_NEWLINE );
+	lineCountCache[ str ] = ( matches ) ? matches.length : 0;
+	return lineCountCache[ str ];
+}
+
+/**
+* Searches for variable in scope matching the expected value.
+*
+* @private
+* @param {Object} scope - VM scope
+* @param {any} expected - expected value to search for
+* @returns {string} variable name or `?` if no match found
+*/
+function findName( scope, expected ) {
+	var keys = objectKeys( scope );
+	var key;
+	var i;
+	for ( i = 0; i < keys.length; i++ ) {
+		key = keys[ i ];
+		if (
+			!contains( SCOPE_DEFAULTS, key ) &&
+			compareValues( scope[ key ], expected ) === null
+		) {
+			return key;
+		}
+	}
+	return '?';
+}
+
+/**
+* Resolves the implementation path relative to a TypeScript declaration file.
+*
+* @private
+* @param {string} filepath - TypeScript declaration file path
+* @param {string} implementationPath - relative path to implementation
+* @returns {string|null} resolved implementation path or null if not found
+*/
+function resolveImplementationPath( filepath, implementationPath ) {
+	var implPath;
+	var baseDir;
+
+	baseDir = dirname( resolve( filepath ) );
+	implPath = resolve( baseDir, implementationPath );
+	if ( existsSync( implPath ) ) {
+		return implPath;
+	}
+	return null;
+}
+
+/**
+* Cleans TSDoc comment by removing comment prefixes.
+*
+* @private
+* @param {string} comment - TSDoc comment
+* @returns {string} cleaned comment
+*/
+function cleanTSDocComment( comment ) {
+	// Remove opening /** and closing */
+	var cleaned = comment.replace( /^\/\*\*/, '' ).replace( /\*\/$/, '' );
+
+	// Remove * at the beginning of lines
+	cleaned = replace( cleaned, RE_COMMENT_PREFIX, '' );
+	return cleaned;
+}
+
+/**
+* Processes a single example code block, finding and validating annotations.
+*
+* @private
+* @param {string} code - example code
+* @param {number} commentIdx - comment index
+* @param {Array} comments - all comments
+* @param {Object} scope - VM execution scope
+* @param {Function} report - error reporting function
+* @param {Object} opts - options object
+* @param {Object} sourceCode - source code object
+*/
+function processExampleCode( code, commentIdx, comments, scope, report, opts, sourceCode ) {
+	var returnAnnotationPattern;
+	var valueStartInAnnotation;
+	var commentStartIdx;
+	var annotationMatch;
+	var annotationStart;
+	var exampleStartIdx;
+	var annotationEnd;
+	var intermediary;
+	var replacement;
+	var exampleIdx;
+	var remaining;
+	var expected;
+	var codeIdx;
+	var actual;
+	var last;
+	var line;
+	var type;
+	var loc;
+	var msg;
+	var arr;
+
+	last = 0;
+	RE_ANNOTATION.lastIndex = 0;
+	try {
```

Comment:
I am not sure why we are enclosing the entirety of the code logic in a `try` block. This is obviously not best practice. I see that the same thing was done in [`jsdoc-doctest`](https://github.com/stdlib-js/stdlib/blob/724283b0bc5d2eb8bf48491c954c6a119330c1f5/lib/node_modules/%40stdlib/_tools/eslint/rules/jsdoc-doctest/lib/main.js#L277); that should also be cleaned up. Otherwise, we are swallowing programmer errors.